source/source.go

```diff
@@ -48,7 +49,12 @@ type Source struct {
 
 // NewSource initialises a new source.
 func NewSource() sdk.Source {
-	return sdk.SourceWithMiddleware(&Source{}, sdk.DefaultSourceMiddleware()...)
+	return sdk.SourceWithMiddleware(&Source{}, sdk.DefaultSourceMiddleware(
+		sdk.SourceWithSchemaExtractionConfig{
+			PayloadEnabled: lang.Ptr(false),
+			KeyEnabled:     lang.Ptr(false),
+		},
```

Comment:
Hmm, why are we doing this?

Reply:
Because  by default the sdk extracts and encodes the record key with a schema. The key provided in the record ([2]) is encoded in a different format than the one expected (map[col1:1]). This mismatch leads to the test failure.

So, disabling schema extraction ensures the key remains in its raw form.

Reply:
Okay we don't need to disable schema extraction for the key.
I have now passed the key as encoded value derived using encodeKey() which converts an id into Avro-style variable-length zigzag encoding as a byte slice.

Now getting the expected key.